Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp

```diff
@@ -702,7 +702,10 @@ namespace AzToolsFramework
                 toPath.ReplaceFilename(newVal.toStdString().c_str());
                 toPath.ReplaceExtension(extension);
             }
-
+            if (fromPath == toPath)
+            {
+                return;
+            }
```

Comment:
nit - worth adding a comment to explain why the check and early return is needed.

Reply:
Will do.